File: GFG/Move Last Element to Front of a Linked List/Move Last Element to Front of a Linked List.py
class Solution:
    def moveToFront(self, head : Optional['Node']) -> Optional['Node']:
        if not head or not head.next: return head
        
        tempNode = Node(0)
        tempNode.next = head
        h1 = head
        while h1.next.next:
            h1 = h1.next
        tempNode.data = h1.next.data
        h1.next = None
        head = tempNode
        return head

File: Move Last Element to Front of a Linked List.cpp
class Solution{
public:
    ListNode *moveToFront(ListNode *head){
        if(!head || !head->next) return head;
        ListNode *node = head;
        ListNode * prev = NULL;
        
        while(node->next != NULL){
            prev = node;
            node = node->next;
        }
        node->next = head;
        prev->next = NULL;
        return node;
    }
};
